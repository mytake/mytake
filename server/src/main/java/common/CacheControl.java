/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import org.jooby.Response;

public class CacheControl {
	public static Response forever(Response res) {
		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
		return res.header("Cache-Control", "public, max-age=31536000, immutable");
	}

	public static Response hour(Response res) {
		// https://support.cloudflare.com/hc/en-us/articles/202775670-Customizing-Cloudflare-s-cache#:~:text=Cache%20additional%20content%20at%20Cloudflare,-Caching%20additional%20content&text=Do%20not%20use%20Cache%20Everything,Cache%20Level%20set%20to%20Bypass.
		return res.header("Cache-Control", "public, max-age=3600, immutable");
	}

	public static Response bypass(Response res) {
		// https://support.cloudflare.com/hc/en-us/articles/202775670-Customizing-Cloudflare-s-cache#:~:text=Cache%20additional%20content%20at%20Cloudflare,-Caching%20additional%20content&text=Do%20not%20use%20Cache%20Everything,Cache%20Level%20set%20to%20Bypass.
		return res.header("Cache-Control", "no-cache");
	}

	public static Response corsAllowAll(Response res) {
		return res.header("Access-Control-Allow-Origin", "*");
	}
}
