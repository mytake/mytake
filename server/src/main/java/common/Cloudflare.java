/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import org.jooby.Result;
import org.jooby.Results;

/** Sends a response with cache-headers which guarantee forever-immutability. */
public class Cloudflare {
	public static Result json(Object toSend) {
		return Results.json(toSend)
				.header("Cache-Control",
						"max-age=31536000", // one year https://stackoverflow.com/a/25201898/1153071
						"public", // any cache may store the response
						"no-transform", // don't muck with it at all
						"immutable" // it will never change at all for sure
				);
	}
}
