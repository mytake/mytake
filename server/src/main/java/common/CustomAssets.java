/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import com.google.common.io.Resources;
import com.google.inject.Binder;
import com.jsoniter.JsonIterator;
import com.jsoniter.any.Any;
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.function.BiFunction;
import java.util.stream.Collectors;
import org.jooby.Env;
import org.jooby.Jooby;
import org.jooby.assets.AssetCompiler;
import org.jooby.rocker.Rockerby;

public class CustomAssets implements Jooby.Module {
	public static void initTemplates(Jooby jooby) {
		// sets variables that the templates need
		jooby.use(new CustomAssets());
		// makes rocker templates work
		jooby.use(new Rockerby());
	}

	private CustomAssets() {}

	static final String _STYLES = "_styles";
	static final String _SCRIPTS = "_scripts";
	static final String FB_APP_ID = "fbappid";

	@Override
	public void configure(Env env, Config conf, Binder binder) throws Throwable {
		env.router().assets("/favicon.ico", "/assets/permanent/favicon-6022c3e42d.ico");

		Config config;
		BiFunction<String, String, String> url;
		String fbAppId = conf.getString("application.fbAppId");
		if (env.name().equals("dev")) {
			config = ConfigFactory.parseResources("assets.dev.conf");
			url = (type, raw) -> "/assets-dev/" + type + raw;
			env.router().assets("/assets-dev/**");
			env.router().assets("/assets/**");
		} else {
			config = ConfigFactory.parseResources("assets.conf");
			byte[] manifest = Resources.toByteArray(CustomAssets.class.getResource("/assets/manifest.json"));
			Map<String, Any> map = JsonIterator.deserialize(manifest).asMap();
			url = (type, raw) -> "/assets/" + type + "/" + Objects.requireNonNull(map.get(raw.substring(1)), "No fingerprinted version of " + raw + ", only has: " + map.keySet());
			env.router().assets("/assets/**");
		}
		AssetCompiler compiler = new AssetCompiler(config);
		// key, style, key, script
		List<String> keyValue = keyValueFor(compiler, url);
		env.router().before((req, rsp) -> {
			for (int i = 0; i < keyValue.size() / 2; ++i) {
				String key = keyValue.get(2 * i);
				String value = keyValue.get(2 * i + 1);
				req.set(key, value);
			}
			req.set(FB_APP_ID, fbAppId);
		});
	}

	static List<String> keyValueFor(AssetCompiler compiler, BiFunction<String, String, String> url) {
		List<String> keyValue = new ArrayList<>(4 * compiler.fileset().size());
		for (String fileset : compiler.fileset()) {
			keyValue.add(fileset + CustomAssets._STYLES);
			keyValue.add(styles(url, compiler, fileset));
			keyValue.add(fileset + CustomAssets._SCRIPTS);
			keyValue.add(scripts(url, compiler, fileset));
		}
		return keyValue;
	}

	private static final String SLASH_SHA_384 = "/sha384-";

	private static String styles(BiFunction<String, String, String> urlMapper, AssetCompiler compiler, String fileset) {
		return compiler.styles(fileset).stream()
				.map(url -> url.startsWith(SLASH_SHA_384) ? url : urlMapper.apply("styles", url))
				.map(CustomAssets::subresourceIntegrityStyle)
				.collect(Collectors.joining("\n"));
	}

	private static String scripts(BiFunction<String, String, String> urlMapper, AssetCompiler compiler, String fileset) {
		return compiler.scripts(fileset).stream()
				.map(url -> url.startsWith(SLASH_SHA_384) ? url : urlMapper.apply("scripts", url))
				.map(CustomAssets::subresourceIntegrityScript)
				.collect(Collectors.joining("\n"));
	}

	private static String subresourceIntegrityStyle(String input) {
		if (input.startsWith(SLASH_SHA_384)) {
			int barIdx = input.indexOf('|');
			String integrity = input.substring(1, barIdx); // 1=remove slash
			String url = input.substring(barIdx + 1);
			return "<link rel=\"stylesheet\" href=\"" + url + "\" integrity=\"" + integrity + "\" crossorigin=\"anonymous\">";
		} else {
			return "<link rel=\"stylesheet\" href=\"" + input + "\">";
		}
	}

	private static String subresourceIntegrityScript(String input) {
		if (input.startsWith(SLASH_SHA_384)) {
			int barIdx = input.indexOf('|');
			String integrity = input.substring(1, barIdx); // 1=remove slash
			String url = input.substring(barIdx + 1);
			return "<script type=\"text/javascript\" src=\"" + url + "\" integrity=\"" + integrity + "\" crossorigin=\"anonymous\"></script>";
		} else {
			return "<script type=\"text/javascript\" src=\"" + input + "\"></script>";
		}
	}
}
