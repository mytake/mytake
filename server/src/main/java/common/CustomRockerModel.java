/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import com.diffplug.common.base.Errors;
import com.fizzed.rocker.RockerTemplate;
import com.fizzed.rocker.RockerTemplateCustomizer;
import com.fizzed.rocker.runtime.DefaultRockerModel;
import com.fizzed.rocker.runtime.StringBuilderOutput;
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import java.util.HashMap;
import java.util.List;
import java.util.function.BiFunction;
import org.jooby.assets.AssetCompiler;

public abstract class CustomRockerModel extends DefaultRockerModel {
	/** Renders this model to a string. */
	public String renderToString() {
		return render(StringBuilderOutput.FACTORY).toString();
	}

	public String renderTest() {
		return super.doRender(null, StringBuilderOutput.FACTORY, new RockerTemplateCustomizer() {
			@Override
			public void customize(RockerTemplate templateRaw) {
				try {
					CustomRockerTemplate template = (CustomRockerTemplate) templateRaw;

					// make assets.conf work
					template.locals = new HashMap<>();
					Config config = ConfigFactory.parseResources("assets.conf");
					AssetCompiler compiler = new AssetCompiler(config);
					BiFunction<String, String, String> url = (type, raw) -> "/assets-dev/" + type + raw;
					List<String> keyValue = CustomAssets.keyValueFor(compiler, url);
					for (int i = 0; i < keyValue.size() / 2; ++i) {
						String key = keyValue.get(2 * i);
						String value = keyValue.get(2 * i + 1);
						template.locals.put(key, value);
					}
				} catch (Throwable e) {
					throw Errors.asRuntime(e);
				}
			}
		}).toString();
	}

}
