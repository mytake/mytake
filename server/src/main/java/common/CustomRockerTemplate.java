/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import com.fizzed.rocker.RockerModel;
import com.google.common.base.Preconditions;
import forms.api.RockerRaw;
import java.util.Iterator;
import java.util.Locale;
import org.jooby.rocker.RequestRockerTemplate;

public abstract class CustomRockerTemplate extends RequestRockerTemplate {
	public CustomRockerTemplate(RockerModel model) {
		super(model);
	}

	/** Returns the single or plural stream as appropriate, based on the size of the iterable. */
	public String singlePlural(String single, String plural, Iterable<?> iterable) {
		Iterator<?> iter = iterable.iterator();
		if (!iter.hasNext()) {
			return plural;
		} else {
			iter.next();
			if (!iter.hasNext()) {
				return single;
			} else {
				return plural;
			}
		}
	}

	/** Returns link tags for the css files with the given fileset. */
	public RockerRaw style(String fileset) {
		String linkTags = (String) locals.get(fileset + CustomAssets._STYLES);
		Preconditions.checkNotNull(linkTags, "No such styles %s, available: %s", fileset, locals.keySet());
		return new RockerRaw().appendRaw(linkTags);
	}

	/** Returns script tags for the js files with the given fileset. */
	public RockerRaw script(String fileset) {
		String scriptTags = (String) locals.get(fileset + CustomAssets._SCRIPTS);
		Preconditions.checkNotNull(scriptTags, "No such scripts %s, available: %s", fileset, locals.keySet());
		return new RockerRaw().appendRaw(scriptTags);
	}

	/** Returns the single or plural stream as appropriate, based on the count. */
	public static String capitalize(String input) {
		if (input.isEmpty()) {
			return input;
		} else {
			return input.substring(0, 1).toUpperCase(Locale.ROOT) + input.substring(1);
		}
	}
}
