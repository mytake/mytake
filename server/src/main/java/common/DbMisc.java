/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import com.diffplug.common.base.Throwing;
import org.jooby.Request;
import org.jooq.DSLContext;
import org.jooq.SelectConditionStep;
import org.jooq.TableField;
import org.jooq.TableRecord;

public class DbMisc {
	public static <T> T transactionFunc(Request req, Throwing.Function<DSLContext, T> transaction) {
		try (DSLContext dslOuter = req.require(DSLContext.class)) {
			return dslOuter.transactionResult(conf -> {
				return transaction.apply(conf.dsl());
			});
		}
	}

	public static void transaction(Request req, Throwing.Consumer<DSLContext> transaction) {
		try (DSLContext dslOuter = req.require(DSLContext.class)) {
			dslOuter.transaction(conf -> {
				transaction.accept(conf.dsl());
			});
		}
	}

	public static <K, R extends TableRecord<R>> R fetchOne(DSLContext dsl, TableField<R, K> keyField, K key) {
		return dsl.selectFrom(keyField.getTable()).where(keyField.eq(key)).fetchOne();
	}

	public static <K, R extends TableRecord<R>, V> V fetchOne(DSLContext dsl, TableField<R, K> keyField, K key, TableField<R, V> valueField) {
		return dsl.selectFrom(keyField.getTable()).where(keyField.eq(key)).fetchOne(valueField);
	}

	public static <K, R extends TableRecord<R>> SelectConditionStep<R> selectWhere(DSLContext dsl, TableField<R, K> keyField, K key) {
		return dsl.selectFrom(keyField.getTable()).where(keyField.eq(key));
	}

}
