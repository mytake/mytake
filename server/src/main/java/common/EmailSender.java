/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import com.diffplug.common.base.Errors;
import com.diffplug.common.base.Throwing;
import com.google.inject.Binder;
import com.typesafe.config.Config;
import java.util.concurrent.Executor;
import java.util.concurrent.Executors;
import org.apache.commons.mail.HtmlEmail;
import org.jooby.Env;
import org.jooby.Jooby;
import org.jooby.Registry;
import org.jooby.mail.CommonsEmail;

/**
 * Sends an email in another thread to avoid blocking.
 */
public class EmailSender {
	private Registry registry;
	private Executor executor;

	private EmailSender(Executor executor) {
		this.executor = executor;
	}

	/** Sends an email. If no "from" is specified, it will be from {@link Emails#TEAM}. */
	public void send(Throwing.Consumer<HtmlEmail> sender) {
		try {
			HtmlEmail htmlEmail = registry.require(HtmlEmail.class);
			htmlEmail.setFrom(Emails.TEAM.email(), Emails.TEAM.name());
			sender.accept(htmlEmail);
			executor.execute(() -> {
				Errors.log().run(htmlEmail::send);
			});
		} catch (Throwable e) {
			throw Errors.asRuntime(e);
		}
	}

	private static class Module implements Jooby.Module {
		@Override
		public void configure(Env env, Config conf, Binder binder) throws Throwable {
			Executor executor = Executors.newCachedThreadPool();
			EmailSender emailSender = new EmailSender(executor);
			binder.bind(EmailSender.class).toInstance(emailSender);
			env.onStart(registry -> {
				emailSender.registry = registry;
			});
		}
	}

	public static void init(Jooby jooby) {
		jooby.use(new CommonsEmail());
		jooby.use(new Module());
	}
}
