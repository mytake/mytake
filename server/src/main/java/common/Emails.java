/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
package common;

import forms.api.RockerRaw;
import java.util.Objects;

public enum Emails {
	TEAM("team", "Team"), TECH("tech", "Technology Team");

	public static final String ADDRESS_1 = "340 S Lemon Ave #3433";
	public static final String ADDRESS_2 = "Walnut, CA 91789";

	private final String email;
	private final String name;

	Emails(String email, String name) {
		this.email = Objects.requireNonNull(email);
		this.name = Objects.requireNonNull(name);
	}

	@Override
	public String toString() {
		throw new UnsupportedOperationException("You probably want linkEmail() or link(String body)");
	}

	public String email() {
		return email + "@mytake.org";
	}

	public String title() {
		return "The MyTake.org " + name;
	}

	public RockerRaw linkEmail() {
		return link(email());
	}

	public RockerRaw link(String body) {
		return RockerRaw.raw("<a href=\"mailto:" + email() + "\">" + body + "</a>");
	}
}
