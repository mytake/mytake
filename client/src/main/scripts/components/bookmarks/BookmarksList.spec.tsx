/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import React from "react";
import renderer from "react-test-renderer";
import { FoundationHarness } from "../../common/foundationTest";
import { BookmarksMode, _bookmarksImpl, _BookmarksWithData } from "./bookmarks";
import BookmarksList from "./BookmarksList";
import { BookmarksResultListProps } from "./BookmarksResultList";

jest.mock("./BookmarksSortBy", () => ({
  __esModule: true,
  default: "BookmarksSortBy",
}));

jest.mock("./BookmarksResultList", () => ({
  __esModule: true,
  default: (props: BookmarksResultListProps) => {
    const { bookmarksResult } = props;
    return (
      <div>BookmarksResultList: {bookmarksResult.factHits.length} hits</div>
    );
  },
}));

test("BookmarksList date happened", () => {
  const result = _bookmarksImpl(
    new _BookmarksWithData(
      FoundationHarness.loadAllFromDisk(),
      BookmarksMode.DateHappened,
      [] // TODO use some data
    )
  );

  const tree = renderer
    .create(
      <BookmarksList
        mode={BookmarksMode.DateBookmarked}
        onModeChange={jest.fn()}
        bookmarksResult={result}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("BookmarksList date bookmarked", () => {
  const result = _bookmarksImpl(
    new _BookmarksWithData(
      FoundationHarness.loadAllFromDisk(),
      BookmarksMode.DateHappened,
      [] // TODO use some data
    )
  );

  const tree = renderer
    .create(
      <BookmarksList
        mode={BookmarksMode.DateBookmarked}
        onModeChange={jest.fn()}
        bookmarksResult={result}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("BookmarksList no results", () => {
  const result = _bookmarksImpl(
    new _BookmarksWithData(
      FoundationHarness.loadAllFromDisk(),
      BookmarksMode.DateHappened,
      []
    )
  );

  const tree = renderer
    .create(
      <BookmarksList
        mode={BookmarksMode.DateBookmarked}
        onModeChange={jest.fn()}
        bookmarksResult={result}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
