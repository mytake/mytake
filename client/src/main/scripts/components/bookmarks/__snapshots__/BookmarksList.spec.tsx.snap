// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarksList date bookmarked 1`] = `
<div
  className="results"
>
  <div
    className="results__inner-container"
  >
    <BookmarksSortBy
      onChange={[MockFunction]}
      selectedOption="DateBookmarked"
    />
    <div>
      BookmarksResultList: 
      0
       hits
    </div>
  </div>
</div>
`;

exports[`BookmarksList date happened 1`] = `
<div
  className="results"
>
  <div
    className="results__inner-container"
  >
    <BookmarksSortBy
      onChange={[MockFunction]}
      selectedOption="DateBookmarked"
    />
    <div>
      BookmarksResultList: 
      0
       hits
    </div>
  </div>
</div>
`;

exports[`BookmarksList no results 1`] = `
<div
  className="results"
>
  <div
    className="results__inner-container"
  >
    <BookmarksSortBy
      onChange={[MockFunction]}
      selectedOption="DateBookmarked"
    />
    <div>
      BookmarksResultList: 
      0
       hits
    </div>
  </div>
</div>
`;
